### DISCUSSION TITLE
Sidebar Insert Filter - hook

### DISCUSSION THREAD

**Discussion by ePiliK** [bootscore/bootscore#786] [Q&A] [answered]
I need a little help.

I have created filters with wpgridbuilder

I would like to hook them via function to the sidebar of the categories and the shop

I tried this way by creating this hook
function add_facet_to_sidebar_hook() {
    if (is_shop() || is_product_category() || is_product_tag()) {
        echo do_shortcode(‘[wpgb_facet id=’1‘ grid=’wpgb-content‘]’)
    }
}
add_action(‘dynamic_sidebar_before’, ‘add_facet_to_sidebar_hook’);

I have a problem it prints it right inside the sidebar as the first element, but I don't know why it also prints it before the menu... (it fills it in twice).

Do we have to use a different hook to call something inside the sidebar?

Thank you very much for your help

atm woocommerce.php
defined('ABSPATH') || exit;

get_header();
?>

  <div id="content" class="site-content <?= apply_filters('bootscore/class/container', 'container', 'woocommerce'); ?> <?= apply_filters('bootscore/class/content/spacer', 'pt-3 pb-5', 'woocommerce'); ?>">
    <div id="primary" class="content-area">

      <main id="main" class="site-main">

        <?php woocommerce_breadcrumb(); ?>

        <?php if (is_shop() or is_product_category() or is_product_tag()): ?> 
          <div class="row"> 
          <?php //echo do_shortcode('[wpgb_facet id="1" grid="wpgb-content"]') ?>
          <div class="<?= apply_filters('bootscore/class/main/col', 'col'); ?>">
              <?php woocommerce_content(); ?> 
            </div> 
             
            <?php get_sidebar(); ?> 
             
          </div> 
        <?php else: ?> 
          <?php woocommerce_content(); ?> 
        <?php endif; ?>
        
      </main>
    </div>
  </div>
<?php
get_footer();

**Comment by crftwrk** [comment, marked as answer]
This is because there is no hook in `sidebar.php`, you have to create one.

Create hook in child's `functions.php`:

```php
function add_facet_to_sidebar_hook() {
  do_action('add_facet_to_sidebar_hook');
}
```

```php
function my_custom_facet_content() {
  if (is_shop() || is_product_category() || is_product_tag()) {
    echo '<div class="alert alert-info">This is my custom facet content!</div>';
  }
}

add_action( 'add_facet_to_sidebar_hook', 'my_custom_facet_content' );
```

Copy `sidebar.php` to your child and insert the hook where you want:

```php
<?php add_facet_to_sidebar_hook(); ?>
```

Solved?

**Reply by ePiliK** [reply]
Thanks so much for the advice. I had created a shortcode but I find your solution better!

**Reply by crftwrk** [reply]
You‘re welcome 👍 You can use a shortcode as well, it‘s just an example how to add a hook to the sidebar.
